.github/workflows/pr-labeler.yml

```diff
@@ -0,0 +1,51 @@
+name: PR Labeler
+on: [pull_request]
+
+jobs:
+  label:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/github-script@v6
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+          const fs = require('fs');
+          const path = require('path');
+
+          function getFileExtensions(files) {
+            return files.map(file => path.extname(file).toLowerCase()).filter(ext => ext !== '');
```

Comment:
Wouldn't this scan the entire codebase? What we want are the changes made in the PR, so we should extract that first?

Reply:
I believe on line 43, the GitHub REST API is being called but specifically for listing pull request files. I don't think it's scanning the entire codebase.